### PR TITLE
Escape & in mailbox names

### PR DESCRIPTION
**Pull request by shaniacht1**

### CODE REVIEW

**Comment by dorsha** [comment]
@shaniacht1 ?

**Comment by dorsha** [comment]
@aviadl / @shaniacht1 - what is the status of this PR?

**Comment by Spivaka** [comment]
@aviadl is this ready for merge?

File: Integrations/integration-Office_EWS.yml [hunk 1]
```diff
@@ -1041,12 +1048,52 @@ script:
...
+            var mailboxes = execute('ews-get-searchable-mailboxes', {}, true);
+            args['mailbox-search-scope'] = dq(mailboxes, 'Envelope.Body.GetSearchableMailboxesResponse.SearchableMailboxes.SearchableMailbox.ReferenceId');
+        } else {
+            if (!Array.isArray(args['mailbox-search-scope'])) {
+                args['mailbox-search-scope'] = args['mailbox-search-scope'].split(';');
+            }
+        }
+        args['mailbox-search-scope'] = args['mailbox-search-scope'].map(function(a) { return a.replace(/(&)/g, '&amp;').trim(); });
+        var slicedMailboxes = [];
+        for (var i=0; i< args['mailbox-search-scope'].length / 15000; i++){
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: Please add small comments above this `for` - IMO its not intuitive

File: Integrations/integration-Office_EWS.yml [hunk 2]
```diff
@@ -1041,12 +1048,52 @@ script:
...
+            data = dq(res, currentCommand.innerPath);
+            if (data){
+                lastRes = res;
+                rawData = rawData.concat(data);
+            }
+        }
+        if(!lastRes){
+            return "No results";
+        }
+        lastRes.Envelope.Body.SearchMailboxesResponse.ResponseMessages.SearchMailboxesResponseMessage.SearchMailboxesResult.Items.SearchPreviewItem = rawData;
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: we should wrap this with `try {} catch` statement
if EWS changes a small part of the xml - it will fail

**Pull request merged**